sphinx_gallery/gen_gallery.py

```diff
@@ -294,7 +294,8 @@ def setup(app):
 
     app.add_stylesheet('gallery.css')
 
-    if 'sphinx.ext.autodoc' in app._extensions:
+    extensions_attr = '_extensions' if hasattr(app, '_extensions') else 'extensions'
```

Comment:
> `if 'sphinx.ext.autodoc' in getattr(app, '_extensions', getattr(app, 'extensions')):`

I do think that this would work, though TBH I don't think it's as clear as just adding an extra line as is done in this PR.